python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json

```diff
@@ -1,7 +1,7 @@
 {
   "title": "\"__init__\" should not return a value",
   "type": "BUG",
-  "status": "ready",
+  "status": "deprecated",
```

Comment:
@guillaume-dequenne-sonarsource @jean-jimbo-sonarsource is this deprecation intentional?

Reply:
It was apparently tagged as "superceded" by S935 2 years ago.
However, we did not implement S935 at the time, so it stayed "ready". I think this change has been triggered by the fact that S935 has now been implemented.
I assume the current implementation of S935 does not actually include `__init__` and the scope of S2734, right?
So probably we could keep the 2 rules and keep S2734 as a special case of S935: this seems to me the less disruptive approach.

Reply:
Yes, the current implementation of S935 does not include `__init__`.
Also, the original formulation of the S935 rspec actually did not explicitly list the functions it targets.

Now, such a listing is included, as I merged it from the former S6658 rspec.

The original JIRA implementation ticket for S935 did list some special functions to be implemented, **including `__init__`**: https://sonarsource.atlassian.net/browse/SONARPY-519

 So for now, it is probably best, to keep S2734. When the time comes when work on [SONARPY-519](https://sonarsource.atlassian.net/browse/SONARPY-519) is continued, then it might be good to merge them.

Reply:
I created an rspec PR to revert the deprecation of S2734 for now, until S935 is extended to cover more special functions: https://github.com/SonarSource/rspec/pull/2774